### DISCUSSION TITLE
ClusterTable not showing the correct associated VM count

### DISCUSSION THREAD

**Discussion by kuzmik** [netbox-community/netbox#5890] [Q&A] [answered]
Hey everyone, I am working on our custom plugin here, and ran into a weird issue that I can't figure out. Basically I created a page that displays all Clusters and VMs owned by a specific Tenant (or TenantGroup). However, the ClusterTable is not displaying the VM count correctly. I created a bug report in netbox itself, but was bounced over here since this is for a custom plugin.

### Environment
* Python version: Python 3.8.5
* NetBox version: 2.10.4

### Steps to Reproduce
1. In my plugins `views.py`, I have the ClusterTable being populated like so (which I basically copied off of other spots in the actual Netbox code):
```
cluster_table = ClusterTable(
    list(tenant.clusters.prefetch_related('group', 'site', 'devices', 'virtual_machines').all()),
    orderable=True,
)
cluster_table.columns.hide('type')
cluster_table.columns.hide('tenant')
cluster_table.columns.hide('device_count')
```
2. In my plugin html file, I have the table being rendered via the `responsive_table.html` partial:
```
{% include 'responsive_table.html' with table=cluster_table %}
```

### Expected Behavior
On page load, I expected to see the table to look something like this (sans the columns that I explicitly hid):
![Screen Shot 2021-02-24 at 16 04 28](https://user-images.githubusercontent.com/4432296/109065843-2d3e8a00-76ba-11eb-9198-e33164d10c0c.png)

### Observed Behavior
The VM count per cluster is 0
![Screen Shot 2021-02-24 at 16 06 54](https://user-images.githubusercontent.com/4432296/109065948-595a0b00-76ba-11eb-8da2-b2b1fb73dc15.png)

Is there something more I have to do to get the counts populated? I didn't see anything when I was digging around the Netbox code.

**Comment by jeremystretch** [comment, marked as answer]
The table is looking for a field named `vm_count` in the data to populate the column. You can add this using `.annotate(vm_count=Count('virtual_machines', distinct=True))` on the queryset.

**Reply by kuzmik** [reply]
🤦 that was it. Thanks as always, Jeremy!
